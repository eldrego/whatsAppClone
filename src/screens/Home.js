import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, Button } from 'react-native';
import styles from '../lib/css/styles';

export default class Home extends Component {
  static navigationOptions = {
    title: 'Home Screen',
  }

  render() {
    const contact = { name: 'John' };
    const onPress = () => {
      this.props.navigation.navigate('chat', contact);
    };

    const title = `Chat with ${contact.name}`;

    return (
      <View style={styles.container}>
        <Button
          onPress={onPress}
          title={title}
        />
      </View>
    );
  }
}

Home.propTypes = {
  navigation: PropTypes.object.isRequired,
};
