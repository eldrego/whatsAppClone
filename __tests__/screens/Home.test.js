import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Home from '../../src/screens/Home';

const navigate = jest.fn();
configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const tree = renderer.create(<Home navigation={{ navigate }} />);
  expect(tree).toMatchSnapshot();
});

it('navigate on press', () => {
  const homeScreen = shallow(<Home navigation={{ navigate }} />);
  homeScreen.find('Button').simulate('press');
  expect(navigate).toBeCalledWith('chat', { name: 'John' });
});

it('has proper navigation options', () => {
  expect(Home.navigationOptions).toMatchSnapshot();
});
