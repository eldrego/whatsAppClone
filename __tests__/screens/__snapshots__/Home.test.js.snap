// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has proper navigation options 1`] = `
Object {
  "title": "Home Screen",
}
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 20,
    }
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Chat with John
      </Text>
    </View>
  </View>
</View>
`;
